packages/katello/rubygem-katello/rubygem-katello.spec

```diff
@@ -215,6 +215,12 @@ cp -pa .%{gem_dir}/* \
 %foreman_bundlerd_file
 %foreman_precompile_plugin -s
 
+mkdir -p %{buildroot}%{foreman_dir}/public/assets/
+for asset in katello bastion bastion_katello
```

Comment:
Isn't `katello` technically already there via the regular asset pipeline? It maps to `%{foreman_assets_plugin}`:
https://github.com/theforeman/foreman-packaging/blob/b38523b6dd1edb81d90e21875275b4ce97f29cff/packages/foreman/foreman/foreman.spec#L837-L838

That is already listed in the `%files` section.

It also makes me wonder (since there are both `bastion` and `bastion_katello` engines) if it actually needs to call `rake assets:precompile[bastion]` and `rake assets:precompile[bastion_katello]`.

Though https://github.com/theforeman/foreman/blob/develop/lib/tasks/plugin_assets.rake does imply it may compile webpack multiple times.

Reply:
Now it's coming back: static assets always worked for Foreman itself but never for plugins. With mod_passenger if a file existed it was used, so served by Apache. Then when it didn't, it fell back to dynamic serving.

So I'm looking for the symlink bit, but I can't find it. Is is required, but that implies all plugins suffer from it. IMHO it's something we must add to the `foreman_precompile_plugin` macro in `foreman.spec`. And also add a macro for the `public/assets/%{gem_name}` path.

Of course Katello will still suffer from this with its 3 engines.

Reply:
That is indeed what I'm saying. It looks like it actually copies them here:
https://github.com/theforeman/foreman-packaging/blob/cdbd9c02a0e5cdb611804f1e53ed05c902d01828/debian/bullseye/foreman/foreman-plugin.mk.in#L12

Reply:
Should be easy to check, tasks, REX, openscap or bootdisk pages should fail in your test setup.

Reply:
Amazing. Now *THAT* is the kind of can if worms I was expecting when starting on that ticket!

Reply:
Some, apologies for the lengthy, examples as this only applies to pages older than and outside of the webpack ecosystem:

```
/usr/share/gems/gems/foreman_openscap-5.1.1/public/assets/foreman_openscap
/usr/share/gems/gems/foreman_openscap-5.1.1/public/assets/foreman_openscap/arf_reports-c063a8acad5b4c51af08f0b41c5936a4e99a7a994c3f9dc25c5ffd3b34cb24ed.js
/usr/share/gems/gems/foreman_openscap-5.1.1/public/assets/foreman_openscap/arf_reports-c063a8acad5b4c51af08f0b41c5936a4e99a7a994c3f9dc25c5ffd3b34cb24ed.js.gz
/usr/share/gems/gems/foreman_openscap-5.1.1/public/assets/foreman_openscap/foreman_openscap.json
/usr/share/gems/gems/foreman_openscap-5.1.1/public/assets/foreman_openscap/load_report-1618505e3c3cacf58e820feb19109e2d5276cf510578b1a98b3765733ff2c8e2.js
/usr/share/gems/gems/foreman_openscap-5.1.1/public/assets/foreman_openscap/load_report-1618505e3c3cacf58e820feb19109e2d5276cf510578b1a98b3765733ff2c8e2.js.gz
/usr/share/gems/gems/foreman_openscap-5.1.1/public/assets/foreman_openscap/period_selector-d2765bb220ed24bb2ee52a4a48acc538e1b25f0dc163a03cae1f490bfe1bec80.js
/usr/share/gems/gems/foreman_openscap-5.1.1/public/assets/foreman_openscap/scap_hosts_show-14e16d0ff4d9dcc14e55255a21366da7e29804ab05e9cc2ab82c38ecdb7adc11.js
/usr/share/gems/gems/foreman_openscap-5.1.1/public/assets/foreman_openscap/period_selector-d2765bb220ed24bb2ee52a4a48acc538e1b25f0dc163a03cae1f490bfe1bec80.js.gz
/usr/share/gems/gems/foreman_openscap-5.1.1/public/assets/foreman_openscap/policy-a5e8d9dfee4d42cdc09c25f55022a9cb5b6b902f3a089ec27a2bf61dd05c20b8.css
/usr/share/gems/gems/foreman_openscap-5.1.1/public/assets/foreman_openscap/policy-a5e8d9dfee4d42cdc09c25f55022a9cb5b6b902f3a089ec27a2bf61dd05c20b8.css.gz
/usr/share/gems/gems/foreman_openscap-5.1.1/public/assets/foreman_openscap/scap_breakdown_chart-8b646e2848e7f3d4b790d03ce914307908164fcfb3f1261ad9384a5ddc1dc5f1.css
/usr/share/gems/gems/foreman_openscap-5.1.1/public/assets/foreman_openscap/policy_dashboard-435a6a3a1b75f1bf11cf88fd785e557542a504845d9696fd3a971ea9dd466c5e.css
/usr/share/gems/gems/foreman_openscap-5.1.1/public/assets/foreman_openscap/scap_hosts_show-14e16d0ff4d9dcc14e55255a21366da7e29804ab05e9cc2ab82c38ecdb7adc11.js.gz
/usr/share/gems/gems/foreman_openscap-5.1.1/public/assets/foreman_openscap/policy_dashboard-435a6a3a1b75f1bf11cf88fd785e557542a504845d9696fd3a971ea9dd466c5e.css.gz
/usr/share/gems/gems/foreman_openscap-5.1.1/public/assets/foreman_openscap/policy_dashboard-fad6bad12bc14aa3a4ab9c629e3ce2f748e42d37f41478b4e1e94e01d86ac3b9.js
/usr/share/gems/gems/foreman_openscap-5.1.1/public/assets/foreman_openscap/policy_dashboard-fad6bad12bc14aa3a4ab9c629e3ce2f748e42d37f41478b4e1e94e01d86ac3b9.js.gz
/usr/share/gems/gems/foreman_openscap-5.1.1/public/assets/foreman_openscap/policy_edit-52b9d0f193fb059965ddef8066f8848f14c9d241da67cbe81fe7ae2f9e034ad7.js
/usr/share/gems/gems/foreman_openscap-5.1.1/public/assets/foreman_openscap/policy_edit-52b9d0f193fb059965ddef8066f8848f14c9d241da67cbe81fe7ae2f9e034ad7.js.gz
/usr/share/gems/gems/foreman_openscap-5.1.1/public/assets/foreman_openscap/reports-47fc56263f2ef961272551c0e5d8078c3e635bf191c1790f1cc65185260472e8.js
/usr/share/gems/gems/foreman_openscap-5.1.1/public/assets/foreman_openscap/reports-47fc56263f2ef961272551c0e5d8078c3e635bf191c1790f1cc65185260472e8.js.gz
/usr/share/gems/gems/foreman_openscap-5.1.1/public/assets/foreman_openscap/reports-c12ad86421f71827f5578fb743228ed67d55cee3c51133a3405a780cd9c4c691.css
/usr/share/gems/gems/foreman_openscap-5.1.1/public/assets/foreman_openscap/reports-c12ad86421f71827f5578fb743228ed67d55cee3c51133a3405a780cd9c4c691.css.gz
/usr/share/gems/gems/foreman_openscap-5.1.1/public/assets/foreman_openscap/scap_breakdown_chart-8b646e2848e7f3d4b790d03ce914307908164fcfb3f1261ad9384a5ddc1dc5f1.css.gz
/usr/share/gems/gems/foreman_remote_execution-5.0.3/public/assets
/usr/share/gems/gems/foreman_remote_execution-5.0.7/public/assets
/usr/share/gems/gems/foreman_remote_execution-5.0.8/public/assets
/usr/share/gems/gems/foreman_remote_execution-5.0.8/public/assets/foreman_remote_execution
/usr/share/gems/gems/foreman_remote_execution-5.0.8/public/assets/foreman_remote_execution/execution_interface-ae136625b481b7cf2a9866e0eb6f266650a3abc1e0fa00f5d568d997157dbd4b.js
/usr/share/gems/gems/foreman_remote_execution-5.0.8/public/assets/foreman_remote_execution/execution_interface-ae136625b481b7cf2a9866e0eb6f266650a3abc1e0fa00f5d568d997157dbd4b.js.gz
/usr/share/gems/gems/foreman_remote_execution-5.0.8/public/assets/foreman_remote_execution/foreman_remote_execution-21b440205a2f7cfd3a94777cf7542292da058d63b4ddf047d6b386b96fd3c548.css
/usr/share/gems/gems/foreman_remote_execution-5.0.8/public/assets/foreman_remote_execution/foreman_remote_execution-21b440205a2f7cfd3a94777cf7542292da058d63b4ddf047d6b386b96fd3c548.css.gz
/usr/share/gems/gems/foreman_remote_execution-5.0.8/public/assets/foreman_remote_execution/foreman_remote_execution.json
/usr/share/gems/gems/foreman_remote_execution-5.0.8/public/assets/foreman_remote_execution/job_templates-c4d01a30d69f5f010a7f2c4b6827e8f5b1907efde003f012cb725071b670c029.js
/usr/share/gems/gems/foreman_remote_execution-5.0.8/public/assets/foreman_remote_execution/job_templates-c4d01a30d69f5f010a7f2c4b6827e8f5b1907efde003f012cb725071b670c029.js.gz
/usr/share/gems/gems/foreman_remote_execution-5.0.8/public/assets/foreman_remote_execution/template_input-a4efe052f49ce014899543761cad73f3da6d96e728772251608d093a890bb9f0.js
/usr/share/gems/gems/foreman_remote_execution-5.0.8/public/assets/foreman_remote_execution/template_input-a4efe052f49ce014899543761cad73f3da6d96e728772251608d093a890bb9f0.js.gz
/usr/share/gems/gems/foreman_remote_execution-5.0.8/public/assets/foreman_remote_execution/template_invocation-71a4116e4ddbee6ef06a0c82efa8b1160e78a570828ca243055c4fc2df6e4fc2.js
/usr/share/gems/gems/foreman_remote_execution-5.0.8/public/assets/foreman_remote_execution/template_invocation-71a4116e4ddbee6ef06a0c82efa8b1160e78a570828ca243055c4fc2df6e4fc2.js.gz
/usr/share/gems/gems/foreman_rh_cloud-5.0.31/public/assets
/usr/share/gems/gems/foreman_rh_cloud-5.0.36/public/assets
/usr/share/gems/gems/foreman_rh_cloud-5.0.39/public/assets
/usr/share/gems/gems/foreman_rh_cloud-5.0.41/public/assets
/usr/share/gems/gems/foreman_rh_cloud-5.0.41/public/assets/foreman_rh_cloud
/usr/share/gems/gems/foreman_rh_cloud-5.0.41/public/assets/foreman_rh_cloud/foreman_rh_cloud.json
/usr/share/gems/gems/foreman-tasks-5.2.1/public/assets
/usr/share/gems/gems/foreman-tasks-5.2.2/public/assets
/usr/share/gems/gems/foreman-tasks-5.2.3/public/assets
/usr/share/gems/gems/foreman-tasks-5.2.3/public/assets/foreman-tasks
/usr/share/gems/gems/foreman-tasks-5.2.3/public/assets/foreman-tasks/foreman-tasks.json
/usr/share/gems/gems/foreman-tasks-5.2.3/public/assets/foreman_tasks
/usr/share/gems/gems/foreman-tasks-5.2.3/public/assets/foreman_tasks/foreman_tasks-5c216788114727b5e1a8ec773b6a9292eb574344e993aad8cb7dd6d655bebc94.js
/usr/share/gems/gems/foreman-tasks-5.2.3/public/assets/foreman_tasks/foreman_tasks-5c216788114727b5e1a8ec773b6a9292eb574344e993aad8cb7dd6d655bebc94.js.gz
/usr/share/gems/gems/foreman-tasks-5.2.3/public/assets/foreman_tasks/foreman_tasks-819977735d4974feaf40cb92589b85cbe3f5b19e5794f8d31debb1501031a83d.css
/usr/share/gems/gems/foreman-tasks-5.2.3/public/assets/foreman_tasks/foreman_tasks-819977735d4974feaf40cb92589b85cbe3f5b19e5794f8d31debb1501031a83d.css.gz
/usr/share/gems/gems/foreman-tasks-5.2.3/public/assets/foreman_tasks/trigger_form-5c216788114727b5e1a8ec773b6a9292eb574344e993aad8cb7dd6d655bebc94.js
/usr/share/gems/gems/foreman-tasks-5.2.3/public/assets/foreman_tasks/trigger_form-5c216788114727b5e1a8ec773b6a9292eb574344e993aad8cb7dd6d655bebc94.js.gz
/usr/share/gems/gems/foreman-tasks-5.2.3/public/assets/foreman_tasks/trigger_form-819977735d4974feaf40cb92589b85cbe3f5b19e5794f8d31debb1501031a83d.css
/usr/share/gems/gems/foreman-tasks-5.2.3/public/assets/foreman_tasks/trigger_form-819977735d4974feaf40cb92589b85cbe3f5b19e5794f8d31debb1501031a83d.css.gz
```

Reply:
Right, I get an loading error on e.g. https://centos8-stream-katello-4-5.tanso.example.com/job_templates/new which says it can't load https://centos8-stream-katello-4-5.tanso.example.com/assets/foreman_remote_execution/template_input-a4efe052f49ce014899543761cad73f3da6d96e728772251608d093a890bb9f0.js

I can work with this. Thanks!

Reply:
https://github.com/theforeman/foreman-packaging/pull/8378 should fix the macros, then fix all the plugin packages -.-

Reply:
This was merged, and this PR is rebased to use those macros.

Would you please have another look? :)